### PR TITLE
packages/microsoft_sqlserver: Fix `HEALTHCHECK`

### LINKED ISSUE #10889: [microsoft_sqlserver]  Failing test daily: system test: errorlog in microsoft_sqlserver.log
- Stack version: 8.16.0-SNAPSHOT
- Package: microsoft_sqlserver
- Failing test: system test: errorlog (variant MSSQL_2019-latest)
- DataStream: log
- Owners:
    - @elastic/obs-infraobs-integrations

Failure:
```
Error: error running package system tests: could not complete test run: could not setup service: service is unhealthy: timeout waiting for healthy container
```

Running docker inspect with the mssql container, the healthcheck for that container fails with this error:
```
                    {
                        "Start": "2024-08-26T18:42:09.028716844+02:00",
                        "End": "2024-08-26T18:42:09.063568887+02:00",
                        "ExitCode": 127,
                        "Output": "/bin/sh: 1: /opt/mssql-tools/bin/sqlcmd: not found\n"
                    }
```

Some of the latest builds where it failed:
- https://buildkite.com/elastic/integrations/builds/15011
- https://buildkite.com/elastic/integrations/builds/14993

### PR DESCRIPTION
**Pull request by shmsr**
## Proposed commit message

Update the bin directory for `sqlcmd` because for 2019-latest the directory has changed and also how it works. Please see [this](https://github.com/elastic/integrations/issues/10889#issuecomment-2310917114) comment and [this](https://github.com/elastic/integrations/issues/10889#issuecomment-2314312934) comment for more details. As this is a change that affects only development, no CHANGELOG entry is needed.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes #10889

### CODE REVIEW

**Comment by shmsr** [comment]
Also, see: https://github.com/elastic/integrations/issues/10889#issuecomment-2314312934


As tests were constantly failing had to debug it as well and track down the issue to:

> ODBC driver version 18, is designed with an ‘encryption-first’ approach, ensuring that utilities like sqlcmd and bcp that utilize the Microsoft ODBC driver will operate under the ‘secure by default’ principle. Users who wish to disable encryption will need to do so explicitly.


So, the fix is to add "-No". Also, improved the HEALTHCHECK a bit.

File: packages/microsoft_sqlserver/_dev/deploy/docker/Dockerfile [hunk 1]
```diff
@@ -6,4 +6,4 @@ ENV SA_PASSWORD='1234_asdf'
 
 # Use the same healthcheck as the Windows version of the image.
 # https://github.com/Microsoft/mssql-docker/blob/a3020afeec9be1eb2d67645ac739438eb8f2c545/windows/mssql-server-windows/dockerfile#L31
-HEALTHCHECK --interval=1s --retries=90 CMD /opt/mssql-tools/bin/sqlcmd -U SA -P ${SA_PASSWORD} -Q "select 1"
+HEALTHCHECK --interval=10s --timeout=3s --start-period=10s --retries=3 CMD /opt/mssql-tools18/bin/sqlcmd -S localhost -U SA -P ${SA_PASSWORD} -No -Q "SELECT 1" || exit 1
```
[the remark this answers is not in the document]
Comment (shmsr) [reply packages/microsoft_sqlserver/_dev/deploy/docker/Dockerfile:9]: @anil-elastic I didn't remove the `retries`, please see the 4th argument for HEALTHCHECK where I have set `retries` to 3. Does that answer your question?

[the remark this answers is not in the document]
Comment (anil-elastic) [reply packages/microsoft_sqlserver/_dev/deploy/docker/Dockerfile:9]: @shmsr yes, thank you for the clarification.

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**